Cargo.toml

```
@@ -1,18 +1,25 @@
 [package]
 authors = ["Andrewmatilde <davis6813585853062@outlook.com>", "Hexilee <i@hexilee.me>"]
 edition = "2018"
-name = "tproxy"
-version = "0.1.0"
+name = "rs-tproxy"
+version = "0.1.1"
 
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
+[[bin]]
+name = "rs-tproxy"
+path = "rs-tproxy-controller/src/main.rs"
+
+[workspace]
+members = ["rs-tproxy-controller", "rs-tproxy-proxy"]
+
 [dependencies]
 anyhow = "1.0"
 clap = "2.33.3"
 futures = "0.3.10"
 http = "0.2.3"
 humantime-serde = "1.0"
-hyper = {version = "0.14.4", features = ["runtime", "client", "server", "http1", "http2", "stream"]}
+hyper = {git = "https://github.com/Andrewmatilde/hyper.git", features = ["runtime", "client", "server", "http1", "http2", "stream", "error_return"]}
```

why need a fork repo here?

---

I have configured hyper , add a pub method called serve_connection_with_parts and add some customed implement.The method will return error instead of return 404 when it comes to a http parse error.

---

Because of this , if user send https or ssh message to the port , proxy will meet a parse error and turn into tcp transfer.